### PR TITLE
Anti alias resize

### PR DESCRIPTION
**Pull request by ducha-aiki**
### Description
Added anti-alias flag to resize and rescale.
Example is here https://github.com/kornia/kornia-examples/blob/master/Resize-Antialias.ipynb
image

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: test/geometry/transform/test_affine.py [hunk 1]
```diff
@@ -28,6 +28,11 @@ def test_downsize(self, device, dtype):
         out = kornia.resize(inp, (3, 1))
         assert out.shape == (1, 3, 3, 1)
 
+    def test_downsizeAA(self, device, dtype):
+        inp = torch.rand(1, 3, 10, 8, device=device, dtype=dtype)
+        out = kornia.resize(inp, (5, 3), antialias=True)
+        assert out.shape == (1, 3, 5, 3)
```
Comment (Borda) [inline test/geometry/transform/test_affine.py:34]: add some value check?

Comment (ducha-aiki) [reply test/geometry/transform/test_affine.py:34]: Apparently our resize as a whole do not have any value check :) But yes, good point

File: kornia/geometry/transform/affwarp.py [hunk 2]
```diff
@@ -508,7 +511,7 @@ def _side_to_image_size(
 
 def resize(input: torch.Tensor, size: Union[int, Tuple[int, int]],
            interpolation: str = 'bilinear', align_corners: Optional[bool] = None,
-           side: str = "short") -> torch.Tensor:
+           side: str = "short", antialias: Optional[bool] = False) -> torch.Tensor:
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:514]:
```suggestion
           side: str = "short", antialias: [bool = False) -> torch.Tensor:
```

File: kornia/geometry/transform/affwarp.py [hunk 3]
```diff
@@ -562,6 +583,7 @@ def rescale(
     factor: Union[float, Tuple[float, float]],
     interpolation: str = "bilinear",
     align_corners: Optional[bool] = None,
+    antialias: Optional[bool] = False
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:586]:
```suggestion
    antialias: bool = False
```

File: kornia/geometry/transform/affwarp.py [hunk 4]
```diff
@@ -523,6 +526,9 @@ def resize(input: torch.Tensor, size: Union[int, Tuple[int, int]],
             https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.interpolate for detail
         side (str): Corresponding side if ``size`` is an integer. Can be one of ``"short"``, ``"long"``, ``"vert"``,
             or ``"horz"``. Defaults to ``"short"``.
+        antialias(bool): if True, then image will be filtered with Gaussian before downscaling.
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:529]:
```suggestion
        antialias (bool): if True, then image will be filtered with Gaussian before downscaling.
```

File: kornia/geometry/transform/affwarp.py [hunk 5]
```diff
@@ -573,6 +595,8 @@ def rescale(
...
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:598]:
```suggestion
        antialias (bool): if True, then image will be filtered with Gaussian before downscaling.
```

File: kornia/geometry/transform/affwarp.py [hunk 6]
```diff
@@ -607,6 +631,8 @@ class Resize(nn.Module):
...
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:634]:
```suggestion
        antialias (bool): if True, then image will be filtered with Gaussian before downscaling.
```

File: kornia/geometry/transform/affwarp.py [hunk 7]
```diff
@@ -619,15 +645,18 @@ class Resize(nn.Module):
     """
 
     def __init__(self, size: Union[int, Tuple[int, int]], interpolation: str = 'bilinear',
-                 align_corners: Optional[bool] = None, side: str = "short") -> None:
+                 align_corners: Optional[bool] = None, side: str = "short",
+                 antialias: Optional[bool] = False) -> None:
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:649]:
```suggestion
                 antialias: bool = False) -> None:
```

File: kornia/geometry/transform/affwarp.py [hunk 8]
```diff
@@ -736,6 +765,8 @@ class Rescale(nn.Module):
             ``"bicubic"``, ``"trilinear"``, or ``"area"``. Default: ``"bilinear"``.
         align_corners(bool): Interpolation flag. Default: None. See :func:`~torch.nn.functional.interpolate` for
             details.
+        antialias(bool): if True, then image will be filtered with Gaussian before downscaling.
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:768]:
```suggestion
        antialias (bool): if True, then image will be filtered with Gaussian before downscaling.
```

File: kornia/geometry/transform/affwarp.py [hunk 9]
```diff
@@ -751,15 +782,18 @@ def __init__(
         self,
         factor: Union[float, Tuple[float, float]],
         interpolation: str = "bilinear",
-        align_corners: Optional[bool] = None
+        align_corners: Optional[bool] = None,
+        antialias: Optional[bool] = False
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:786]:
```suggestion
        antialias: bool = False
```

File: kornia/geometry/transform/affwarp.py [hunk 10]
```diff
@@ -751,15 +782,18 @@ def __init__(
...
     ) -> None:
         super().__init__()
         self.factor: Union[float, Tuple[float, float]] = factor
         self.interpolation: str = interpolation
         self.align_corners: Optional[bool] = align_corners
+        self.antialias: Optional[bool] = antialias
```
Comment (edgarriba) [inline kornia/geometry/transform/affwarp.py:792]:
```suggestion
        self.antialias: bool = antialias
```

**Review by edgarriba — CHANGES_REQUESTED** [review]
docs and typing typos

**Review by edgarriba — APPROVED** [review]
